core/src/test/scala/org/apache/spark/shuffle/sort/IndexShuffleBlockResolverSuite.scala

```diff
@@ -133,4 +133,65 @@ class IndexShuffleBlockResolverSuite extends SparkFunSuite with BeforeAndAfterEa
     }
     assert(firstByte2(0) === 2)
   }
+
+  test("SPARK-23253: index files should be created properly") {
```

Comment:
thanks for adding this, but actually I'm not sure this is covering any cases in the previous test, is it?  I was thinking of just adding something to read the actual index file, and make sure it had the right values to go with the update to the data file (or no updates in some cases).

you may have added a couple more asserts than the original test -- if so, maybe they can just be added to the original?

Reply:
+1 we can add check the index file in the original test case.